hail/src/main/scala/is/hail/utils/Graph.scala

```diff
@@ -34,16 +34,16 @@ object Graph {
   }
 
   def pyMaximalIndependentSet(edgesIR: IR, nodeTypeStr: String, tieBreaker: Option[String]): IR = {
-    val nodeType = IRParser.parseType(nodeTypeStr)
+    val nodeType = IRParser.parsePType(nodeTypeStr)
     
     val edges = Interpret[IndexedSeq[Row]](edgesIR).toArray
 
-    val resultType = TSet(nodeType)
+    val resultType = PSet(nodeType)
     val result = maximalIndependentSet(edges, nodeType, tieBreaker)
-    Literal(resultType, result)
+    Literal(resultType.virtualType, result)
   }
 
-  def maximalIndependentSet(edges: Array[Row], nodeType: Type, tieBreaker: Option[String]): Set[Any] = {
+  def maximalIndependentSet(edges: Array[Row], nodeType: PType, tieBreaker: Option[String]): Set[Any] = {
```

Comment:
take a Type and construct a canonical PType in the body here -- the data is being carried around as safe java objects (the `Array[Row]`) so the ptype isn't a necessary piece of state at this point.